src/scenarios/utils.py

```diff
@@ -1,5 +1,29 @@
+from ipaddress import IPv4Address, IPv6Address, ip_address
+from kubernetes import client, config
```

Comment:
I don't think the utils file should be importing backend stuff. I feel like this dependency indicates that `get_service_ip()` is not a utility and should probably be implemented in `kubernetes_backend.py`

Reply:
We could make `get_service_ip` into an abstract method similar to [`get_tank_ipv4`](https://github.com/bitcoin-dev-project/warnet/blob/725c1f1a5422e3f5973122ac7b8eab9f23780029/src/backends/backend_interface.py#L106). Then rename it to `resolve_tank_dns` to make it less k8s specific.

Reply:
yeah or even use @property to get (and then cache) the IP:

```python
    @property
    def ipv4(self):
        if self._ipv4 is None:
            self._ipv4 = generate_ipv4_addr(self.warnet.subnet)
        return self._ipv4
```